Pcap++/src/DpdkDevice.cpp

```diff
@@ -929,11 +943,10 @@ int DpdkDevice::dpdkCaptureThreadMain(void *ptr)
 
 		if (likely(pThis->m_OnPacketsArriveCallback != NULL))
 		{
-			MBufRawPacket rawPackets[numOfPktsReceived];
+			MBufRawPacketArrayGuard sg(rawPackets, numOfPktsReceived);

```

Comment:
I'm not sure why you need this new class `MBufRawPacketArrayGuard`. Couldn't you just put `MBufRawPacket rawPackets[MAX_BURST_SIZE]` instead?

If the reason is being more efficient (and avoid creating this array), maybe it's better to go over the elements of the array and call to `clear()` for each one, which is exactly what `MBufRawPacketArrayGuard` does in its d'tor?

Reply:
Class “MBufRawPacketArrayGuard” is to enforce the RAII mechanism, but MBufRawPacket's  d'tor would have released resources(mbuf).
I'm not sure MBufRawPacketArrayGuard is necessary if put MBufRawPacket rawPackets[numOfPktsReceived] instead?

Reply:
@seladb @shidaoxuan There are two ways that execution may go from point of callback call:
1. Normal execution
2. Exception is thrown by user provided callback

In either of them resources of packets must be freed because they are allocated in some pages outside of process memory that will be automatically released by the system when process ends (a.e. systemwide memory leak is possible).  
Let us now count the amount of work to be done on every proposed solution:
1. ```MBufRawPacket rawPackets[numOfPktsReceived]``` inside ```if``` clause: UB in C++ but common sense tells that at least ```numOfPktsReceived``` constructors and destructors will be called;
2. ```MBufRawPacket rawPackets[MAX_BURST_SIZE]``` inside ```if``` clause: ```MAX_BURST_SIZE``` constructors and destructors will be called every time -> the overhead is ```MAX_BURST_SIZE - numOfPktsReceived``` on every burst received;
3. The RAII approach: ```MAX_BURST_SIZE``` constructors and destructors are called only once (when entering and leaving ```dpdkCaptureThreadMain```) + exactly ```numOfPktsReceived``` calls to ```MBufRawPacket::clear``` on either of execution paths described above. The small overhead of constructor and destructor of ```MBufRawPacketArrayGuard``` is negated by compiler on optimisation (code may be inlined).  

@seladb @shidaoxuan Please let me hear your opinion on my arguments.

Reply:
@echo-Mike I agree that may be the most efficient, but the overhead of creating `MBufRawPacket` instances x`MAX_BURST_SIZE - numOfPktsReceived` times for every burst isn't that bad and makes the solution much cleaner and easier to understand. I'd go with option 2 (`MBufRawPacket rawPackets[MAX_BURST_SIZE]`)

Reply:
Hi @echo-Mike , did you see my previous message? Please let me know if you agree so we can merge to PR to the main branch